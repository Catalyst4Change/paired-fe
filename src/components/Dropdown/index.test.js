import React from 'react';
import { shallow } from 'enzyme';
import { Dropdown } from './index';

describe('Dropdown', () => {
  let wrapper;
  const mockOptions = ['1', '2', '3', '4'];
  const mockLabel = 'Module';
  const mockHandleChange = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Dropdown
        options={mockOptions}
        label={mockLabel}
        handleChange={mockHandleChange}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChangeOptions when an option dropdown is selected', () => {
    const mockEvent = { name: mockLabel, value: mockOptions[0] };
    wrapper.find('select').simulate('change', mockEvent);
    expect(mockHandleChange).toHaveBeenCalledWith(mockEvent);
  });
});
