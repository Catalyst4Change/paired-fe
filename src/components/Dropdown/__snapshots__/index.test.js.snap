// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should match the snapshot 1`] = `
<div
  className="Dropdown--div"
>
  <label
    className="Dropdown--label"
    htmlFor="Module"
  >
    Module
  </label>
  <select
    className="Dropdown--select"
    name="Module"
    onChange={[Function]}
  >
    <option
      value=""
    >
      Please choose an option
    </option>
    <option
      key="option1"
      value="1"
    >
      1
    </option>
    <option
      key="option2"
      value="2"
    >
      2
    </option>
    <option
      key="option3"
      value="3"
    >
      3
    </option>
    <option
      key="option4"
      value="4"
    >
      4
    </option>
  </select>
</div>
`;
